training/src/main/java/wolox/training/models/User.java

```diff
@@ -0,0 +1,89 @@
+package wolox.training.models;
+
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import java.time.LocalDate;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+import javax.persistence.Table;
+import javax.persistence.Id;
+import wolox.training.exceptions.BookAlreadyOwnedException;
+
+@Entity
+@Table(name = "users")
+public class User {
+
+    @Id
+    @GeneratedValue
+    private int Id;
+
+    @Column(nullable = false)
+    private String username;
+
+    @Column(nullable = false)
+    private String name;
+
+    @Column(nullable = false)
+    private LocalDate birthdate;
+
+    @ManyToMany(cascade = CascadeType.ALL)
```

Comment:
This cascade type is the used on the baeldung example but is not the right one for this case. This is because, for example, it will cause that the remove of an user provokes the remove of a book
To read more about the Cascade types maybe this [link](https://www.baeldung.com/jpa-cascade-types) could be useful

Reply:
Here is some useful explanation about this topic:

By saying ALL, you are saying: Persist, Merge, Refresh, Delete and Detach.

Detach is only so that something get's detached along the cascade when you detach an object from the persistence context manually. This almost never happens, so we won't focus on this point.

**Persist** says: If you persist the Users instance to the DB, then persist every item that this collection has in the DB as well.
**Delete** says: If you delete the Users instance from the DB, then delete every item that this collection has in the DB as well.
**Merge** says: If you changed the in-memory values of the collection, then merge this changes to the DB so these are reflected there.
**Refresh** says: If something changed in the database values of the collection, then refresh those changes in the in-memory values.